spinalcordtoolbox/deepseg_sc/core.py

```diff
@@ -111,38 +111,28 @@ def find_centerline(algo, image_fname, contrast_type, brain_bool, folder_output,
                       z_max=z_max if brain_bool else None)
 
     elif algo == 'viewer':
-        im_labels = _call_viewer_centerline(Image(image_fname))
-        im_centerline, arr_centerline, _, _ = get_centerline(im_labels, param=ParamCenterline())
-        centerline_filename = sct.add_suffix(image_fname, "_ctr")
-        im_centerline.save(centerline_filename)
+        im_labels = _call_viewer_centerline(im)
+        im_ctl, _, _, _ = get_centerline(im_labels, param=ParamCenterline())
 
     elif algo == 'file':
-        centerline_filename = sct.add_suffix(image_fname, "_ctr")
-        # Re-orient the manual centerline
-        Image(centerline_fname).change_orientation('RPI').save(centerline_filename)
+        im_ctl = Image(centerline_fname)
+        im_ctl.change_orientation('RPI')
 
     else:
         logger.error('The parameter "-centerline" is incorrect. Please try again.')
         sys.exit(1)
 
-    if algo != 'cnn':
-        logger.info("Resample the image to 0.5x0.5 mm in-plane resolution...")
-        fname_res = sct.add_suffix(image_fname, '_resampled')
-        input_resolution = Image(image_fname).dim[4:7]
-        new_resolution = 'x'.join(['0.5', '0.5', str(input_resolution[2])])
-
-        resampling.resample_file(image_fname, fname_res, new_resolution, 'mm', 'linear', verbose=0)
-
-        resampling.resample_file(centerline_filename, centerline_filename, new_resolution, 'mm', 'linear', verbose=0)
+    # TODO: for some reason, when algo == 'file', the absolutepath is changed to None out of the method find_centerline
+    im_ctl.absolutepath = ctl_absolute_path
```

Comment:
For some reason, when algo == 'file', I could not set the absolute path of `im_ctl`, while it works with `svm` and `viewer`.
@jcohenadad : Any ideas?

Reply:
no quick idea unfortunately. I will have to run the debugger to understand the cause of the problem.

Reply:
btw: an alternative would be to specify file name upon saving:
~~~
im_ctl.save(ctl_absolute_path)
~~~